src/main/webapp/js/richTextEditor.js

```diff
@@ -39,15 +89,129 @@ var richTextEditorBuilder = {
+            },
+            success: function(data) {
+                setTimeout(function() {
+                    if (data.isError) {
+                        setErrorMessage(data.ajaxStatus);
+                    } else {
+                        $('#' + additionalParams.uploadImageId + 'Form').attr('action', data.nextUploadUrl);
+                        setStatusMessage(data.ajaxStatus);
+                        submitImageUploadFormAjax();
+                    }
+                }, 500);
+            }
+        });
+    };
+
+    var submitImageUploadFormAjax = function() {
+        var uploadImageId = additionalParams.uploadImageId;
+        var formData = new FormData($('#' + uploadImageId + 'Form')[0]);
+
+        $.ajax({
+            type: 'POST',
+            enctype: 'multipart/form-data',
+            url: $('#' + uploadImageId + 'Form').attr('action'),
+            data: formData,
+            // Options to tell jQuery not to process data or worry about content-type.
+            cache: false,
+            contentType: false,
+            processData: false,
+
+            beforeSend: function() {
+                showUploadingGif();
+            },
+            error: function() {
+                setErrorMessage('Image upload failed, please try again.');
+                clearUploadFileInfo();
+            },
+            success: function(data) {
+                setTimeout(function() {
+                    if (data.isError) {
+                        setErrorMessage(data.ajaxStatus);
+                    } else if (data.isFileUploaded) {
+                        url = location.origin + data.fileSrcUrl;
+                        callbackFunction(url, { alt: 'PLACEHOLDER_IMAGE_UPLOAD_ALT_TEXT' });
+                        setStatusMessage(data.ajaxStatus, StatusType.SUCCESS);
+                    } else {
+                        setErrorMessage(data.ajaxStatus);
+                    }
+                }, 500);
+                
+            }
 
-    initEditor: function(selector, opts) {
-        tinymce.init($.extend(this.getDefaultConfiguration(), {
-            selector: selector
-        }, opts));
+        });
+        clearUploadFileInfo();
+    };
+
+    var injectImageUploadForm = function() {
+        var uploadImageId = additionalParams.uploadImageId;
+        var uploadImageInputName = additionalParams.uploadImageInputName;
+
+        if ($('#' + uploadImageId + 'Form').length) {
+            return;
+        }
+
+        $('body').append($('<div id="uploadFileBlock" style="display: none;"></div>'));
+        $('#uploadFileBlock').append(
+                '<form id="' + uploadImageId + 'Form" action="" method="POST" enctype="multipart/form-data">');
+        $('#' + uploadImageId + 'Form').append('<span id="' + uploadImageId + 'Input" />');
+        $('#' + uploadImageId + 'Input').append(
+                '<input type="file" id="' + uploadImageId + '" name="' + uploadImageInputName + '" />');
```

Comment:
Closing with `/>` is against our style guide. Either close it with `>` for void elements, or use proper closing tags `</elementname>` for non-void elements. Unless XHTML is required in this operation?